test/https.ts

```diff
@@ -34,3 +107,67 @@ test('http2', async t => {
 	t.is(headers[':status'], 200);
 	t.is(typeof body, 'string');
 });
+
+test.serial('deprecated rejectUnauthorized', withServer, async (t, server, got) => {
+	server.get('/', (_request, response) => {
+		response.end('ok');
+	});
+
+	await new Promise(resolve => {
+		process.once('warning', warn => {
```

Comment:
Would be better to use `p-event`.

Reply:
```js
pEvent(process, 'warning').then(warn => {
  t.is(warn.name, 'DeprecationWarning');
  resolve();
});
```
That's the best I can do without breaking the logic of the test. I can't remove that promise.

I can achieve something by using `Promise.race`, but it's going to be harder to read.

Reply:
At least do:

```js
(async () => {
	const warning = await pEvent(process, 'warning');
	t.is(warning.name, 'DeprecationWarning');
	resolve();
})();
```

Reply:
The `async () => ` function wrapper is not required as the test itself is async.

Reply:
> The `async () => ` function wrapper is not required as the test itself is async.

It's required because the test must continue in order to make the request, then when the event received is checked.